script.js

```diff
@@ -1,4 +1,704 @@
+    suit: "spades",
+    rank: 8,
+  },
+  {
+    name: "9",
+    suit: "spades",
+    rank: 9,
+  },
+  {
+    name: "10",
+    suit: "spades",
+    rank: 10,
+  },
+  {
+    name: "jack",
+    suit: "spades",
+    rank: 11,
+  },
+  {
+    name: "queen",
+    suit: "spades",
+    rank: 12,
+  },
+  {
+    name: "king",
+    suit: "spades",
+    rank: 13,
+  },
+];
+
+var getRandomIndex = function (max) {
+  return Math.floor(Math.random() * max);
+};
+
+// Shuffle the elements in the cardDeck array
+var shuffleCards = function (cardDeck) {
+  // Loop over the card deck array once
+  var currentIndex = 0;
+  while (currentIndex < cardDeck.length) {
+    // Select a random index in the deck
+    var randomIndex = getRandomIndex(cardDeck.length);
+    // Select the card that corresponds to randomIndex
+    var randomCard = cardDeck[randomIndex];
+    // Select the card that corresponds to currentIndex
+    var currentCard = cardDeck[currentIndex];
+    // Swap positions of randomCard and currentCard in the deck
+    cardDeck[currentIndex] = randomCard;
+    cardDeck[randomIndex] = currentCard;
+    // Increment currentIndex
+    currentIndex = currentIndex + 1;
+  }
+  // Return the shuffled deck
+  return cardDeck;
+};
+
+var shuffledDeck = shuffleCards(deck);
+
+// ******* MAIN Function *******//
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  var myOutputValue = "";
+
+  //First Click//
+  if (currentGameMode == gameMode_Start) {
+    //create the game deck
+    var gameDeck = deck;
+    console.log(gameDeck);
+    // deal 2 cards to player and dealer each
+    playerHand.push(gameDeck.pop());
+    playerHand.push(gameDeck.pop());
+    dealerHand.push(gameDeck.pop());
+    dealerHand.push(gameDeck.pop());
+
+    console.log("player hand");
+    console.log(playerHand);
+    console.log("dealer hand");
+    console.log(dealerHand);
+
+    // update the game mode
+    currentGameMode = gameMode_DrawnCards;
+
+    // show updated output message
+    myOutputValue =
+      image1 +
+      "<br>Everyone already draw their cards. Click 'submit' to see the result of cards";
+    return myOutputValue;
+  }
+
+  //Second Click//
+  if (currentGameMode == gameMode_DrawnCards) {
+    // playerHand = [
+    //   { name: "queen", suit: "clubs", rank: 12 },
+    //   { name: "ace", suit: "diamonds", rank: 1 },
+    // ];
+    // dealerHand = [
+    //   { name: "ace", suit: "spades", rank: 1 },
+    //   { name: 10, suit: "spades", rank: 10 },
+    // ];
+    // console.log(currentGameMode);
+    var playerHasBlackJack = checkForBlackJack(playerHand);
+    var dealerHasBlackJack = checkForBlackJack(dealerHand);
+
+    // console.log("player has blackjack? ");
+    // console.log(playerHasBlackJack);
+    // console.log("dealer has blackjack?");
+    // console.log(dealerHasBlackJack);
+
+    // check whether blackjack situation or not
+    if (playerHasBlackJack == true || dealerHasBlackJack == true) {
```

Comment:
you should almost never need to a check such as == true

the variable itself is already a boolean so take advantage of that.

```js
if (playerHasBlackJack || dealerHasBlackJack)
```